apps/bajour/src/components/frage-des-tages/fdt-poll-block.tsx

```diff
@@ -140,16 +139,18 @@ export const FdtPollBlock = ({poll}: {poll?: ApiV1.PollBlock['poll']}) => {
 
   const autoVote = useCallback(async () => {
     const answerId = router.query.answerId as string
-    if (answerId) {
-      callAction(async () => {
-        await vote({
-          variables: {
-            answerId
-          }
-        })
-      })()
+    if (!answerId || !poll?.id) {
+      return
     }
-  }, [callAction, router.query.answerId, vote])
+    await vote(
+      {
+        variables: {
+          answerId
+        }
+      },
+      poll.id
+    )
+  }, [router.query.answerId])
```

Comment:
Pretty sure the linter complains here, please fix this!

Reply:
Linter didn't complain. Just let run again to make sure. Why should it fail IYO?

Reply:
Linter does complain but throws a warning. The effect has a bug without all dependencies because if a dependency changes, the effect doesnt run

Reply:
ah yeah. having all dependencies lead to endless automatically voting. So, refactoring will be necessary.

Reply:
memoizing as in the other comment stated would fix that as every render the function is recreated, which causes the effect to run, which causes a rerender which causes the function to be recreated etc.